cmd/merge/transactions.go

```diff
@@ -183,3 +211,58 @@ func mergeTransactions(cCtx *cli.Context) error {
 	log.Infof("Finished processing CSV files, wrote %s transactions", printer.Sprintf("%d", cntTxWritten))
 	return nil
 }
+
+func updateInclusionStatus(log *zap.SugaredLogger, checkNodeURIs []string, txs map[string]*common.TxSummaryEntry) (err error) {
+	// Load inclusion status for all transactions
+	workers := numRPCWorkers
+	txC := make(chan *common.TxSummaryEntry, 2000000)
+	respC := make(chan error)
+	blockCache := NewBlockCache()
+
+	// start geth workers
+	for _, checkNodeURI := range checkNodeURIs {
+		for i := 0; i < workers; i++ {
+			w := NewTxUpdateWorker(log, checkNodeURI, txC, respC, blockCache)
+			go w.start()
+		}
+	}
+
+	// send tx to worker
+	inclusionCheckStart := time.Now().UTC()
+	log.Info("Loading inclusion status - sending to workers...")
+	for _, entry := range txs {
```

Comment:
This will write all txs to the channel before processing will start and if there are more than 2kk txs it will block.

Alternative is to launch work pusher(code that writes txs), work consumer(code that reads err) in a goroutine too (in addition to workers). This way it will never block and txs will start processing asap without waiting to write all of them to channel.